internal/services/signalr/web_pubsub_network_acl_resource.go

```diff
@@ -148,85 +123,86 @@ func resourceWebpubsubNetworkACL() *pluginsdk.Resource {
-			return fmt.Errorf("when `default_action` is `Allow` for `public_network`, `allowed_request_types` cannot be specified")
-		} else if defaultAction == webpubsub.ACLActionDeny && len(*networkACL.PublicNetwork.Deny) != 0 {
-			return fmt.Errorf("when `default_action` is `Deny` for `public_network`, `denied_request_types` cannot be specified")
-		}
+	if defaultAction == webpubsub.ACLActionAllow && networkACL.PublicNetwork.Allow != nil && len(*networkACL.PublicNetwork.Allow) != 0 {
+		return fmt.Errorf("when `default_action` is `Allow` for `public_network`, `allowed_request_types` cannot be specified")
+	} else if defaultAction == webpubsub.ACLActionDeny && len(*networkACL.PublicNetwork.Deny) != 0 {
+		return fmt.Errorf("when `default_action` is `Deny` for `public_network`, `denied_request_types` cannot be specified")
+	}
 
-		if networkACL.PrivateEndpoints != nil {
-			for _, privateEndpoint := range *networkACL.PrivateEndpoints {
-				if privateEndpoint.Allow != nil && privateEndpoint.Deny != nil && len(*privateEndpoint.Allow) != 0 && len(*privateEndpoint.Deny) != 0 {
-					return fmt.Errorf("`allowed_request_types` and `denied_request_types` cannot be set together for `private_endpoint`")
-				}
+	if networkACL.PrivateEndpoints != nil {
+		for _, privateEndpoint := range *networkACL.PrivateEndpoints {
+			if privateEndpoint.Allow != nil && privateEndpoint.Deny != nil && len(*privateEndpoint.Allow) != 0 && len(*privateEndpoint.Deny) != 0 {
+				return fmt.Errorf("`allowed_request_types` and `denied_request_types` cannot be set together for `private_endpoint`")
+			}
 
-				if defaultAction == webpubsub.ACLActionAllow && privateEndpoint.Allow != nil && len(*privateEndpoint.Allow) != 0 {
-					return fmt.Errorf("when `default_action` is `Allow` for `private_endpoint`, `allowed_request_types` cannot be specified")
-				} else if defaultAction == webpubsub.ACLActionDeny && privateEndpoint.Deny != nil && len(*privateEndpoint.Deny) != 0 {
-					return fmt.Errorf("when `default_action` is `Deny` for `private_endpoint`, `denied_request_types` cannot be specified")
-				}
+			if defaultAction == webpubsub.ACLActionAllow && privateEndpoint.Allow != nil && len(*privateEndpoint.Allow) != 0 {
+				return fmt.Errorf("when `default_action` is `Allow` for `private_endpoint`, `allowed_request_types` cannot be specified")
+			} else if defaultAction == webpubsub.ACLActionDeny && privateEndpoint.Deny != nil && len(*privateEndpoint.Deny) != 0 {
+				return fmt.Errorf("when `default_action` is `Deny` for `private_endpoint`, `denied_request_types` cannot be specified")
 			}
 		}
-		existing.Properties.NetworkACLs = &networkACL
 	}
+	payload.Properties.NetworkACLs = &networkACL
 
-	future, err := client.Update(ctx, existing, id.ResourceGroup, id.WebPubSubName)
-	if err != nil {
+	if err := client.UpdateThenPoll(ctx, *id, payload); err != nil {
```

Comment:
We're getting test failures here. I did some digging and the body being sent to the api is nil. 

```
Error: updating Network ACL configuration for "Web Pub Sub (Subscription: \"*******\"\nResource Group Name: \"acctestRG-webpubsub-XXXXXXX\"\nResource Name: \"acctestRG-webpubsub-XXXXXX\")": performing Update: webpubsub.WebPubSubClient#Update: Failure sending request: StatusCode=0 -- Original Error: Code="InvalidResource" Message="The resource definition is invalid."
```